packages/blade/src/components/BottomNav/types.ts

```diff
@@ -0,0 +1,65 @@
+import type { StyledPropsBlade } from '~components/Box/styledProps';
+import type { IconComponent } from '~components/Icons';
+import type { TestID } from '~utils/types';
+
+type BottomNavProps = {
+  /**
+   * children slot of BottomNav, accepts BottomNavItem
+   */
+  children: React.ReactElement[];
+
+  /**
+   * zIndex of BottomNav
+   *
+   * @default 100
+   */
+  zIndex?: number;
+} & StyledPropsBlade &
+  TestID;
+
+type BottomNavItemProps = {
+  /**
+   * Title text of the BottomNavItem
+   */
+  title: string;
+
+  /**
+   * Icon rendered above the title.
+   *
+   * Accepts icon component from blade
+   */
+  icon: IconComponent;
+
+  /**
+   * href property of link
+   *
+   * maps to `to` property when react router is being used
+   */
+  href?: string;
+
+  /**
+   * as prop to pass ReactRouter's Link component.
+   *
+   * ```jsx
+   * import { NavLink } from 'react-router-dom';
+   *
+   * <BottomNavItem as={Link} />
+   * ```
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  as?: React.ComponentType<any>;
+
+  /**
+   * Active state of the BottomNavItem.
+   *
+   * Use this to set the current page's active state using react router
+   */
+  isActive?: boolean;
+
+  /**
+   * onClick handler of BottomNavItem
+   */
+  onClick?: () => void;
```

Comment:
Expose onClick `event` and also expose `ref` from the BottomNavItem. `(event) => void`

React router depends on ref to inject few attributes into DOM, and ReactRouterV6 requires the onClick's event.

Reply:
We're using `as={NavLink}` for React Router so it won't be used for routing but yup I'll expose event either ways

Reply:
Sure, 

Ideally our components should be composable with any method. for example react router v6 expects composition via their hook and passing props like ref etc. 

https://reactrouter.com/en/main/upgrading/v5#remove-link-component-prop